### PR TITLE
Missing push notifications on login

### LINKED ISSUE #6891: Push messages not received on login when a unified push distributor is installed
### Steps to reproduce

1. Have a unified push distributor installed eg ntfy
2. With a fresh install, login
3. Select Google services when prompted
4. Exit the app, notice no push messages are received

#### What did you expect?
The receive push notifications

#### What happened instead?
Firebase push notifications are received but they're filtered out by `UnifiedPush`

### Your phone model

_No response_

### Operating system version

_No response_

### Application version and app store

1.4.32

### Homeserver

_No response_

### Will you send logs?

No

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by ouchadam**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

- We were saving the default distributor before asking the user, which meant when the user selected the same option it was skipped from being registered
- Only _unregisters_ during the force flow, otherwise we'll crash due to no app being registered

## Motivation and context

Fixes #6891 relates to #6782 

## Screenshots / GIFs

No UI changes

## Tests

- Have a unified push distributor installed eg ntfy
- With a fresh install, login 
- Select `Google services` when prompted
- Exit the app, notice no push messages are received  

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 28

### CODE REVIEW

File: vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt [hunk 1]
```diff
@@ -92,16 +92,21 @@ class UnifiedPushHelper @Inject constructor(
                 return@launch
             }
 
-            // By default, use internal solution (fcm/background sync)
-            UnifiedPush.saveDistributor(context, context.packageName)
```
Comment (ouchadam) [inline vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:96]: preemptively setting the distributor without registering the app causes the user selection to be skipped if it matches https://github.com/vector-im/element-android/pull/6890/files#diff-d0af9539e5621c00bfa7f4315721ab35c43af91f06e81d8f8a8b097d51a45487L152

File: vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt [hunk 2]
```diff
@@ -92,16 +92,21 @@ class UnifiedPushHelper @Inject constructor(
...
                 UnifiedPush.registerApp(context)
                 onDoneRunnable?.run()
             } else {
-                openDistributorDialogInternal(activity, pushersManager, onDoneRunnable, distributors, !force, !force)
+                openDistributorDialogInternal(
+                        activity = activity,
+                        pushersManager = pushersManager,
+                        onDoneRunnable = onDoneRunnable,
+                        distributors = distributors,
+                        unregisterFirst = force,
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:107]: Just to highlight that `unregisterFirst` parameter has been changed from `!force` to `force`.

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

File: vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt (hunk 2 above)
Comment (ouchadam) [reply vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:107]: correct, it's intentional, the unified push connector crashes when attempting to remove a "null" token

Comment (p1gp1g) [reply vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:107]: @ouchadam : I've seen your comment and I can't find a way to crash it with a null token. Anything relevant to fix your issue in the lib is welcome

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:107]: using the parent commit for this PR a251002808921a1da44556795b073a7449257c27 and commenting out the default `UnifiedPush.saveDistributor(context, context.packageName)` https://github.com/vector-im/element-android/pull/6890/commits/0a2ce698becd39abddc85ca4263ea57a211ca052#diff-d0af9539e5621c00bfa7f4315721ab35c43af91f06e81d8f8a8b097d51a45487L96 

when unregistering during the dialog flow, this crash happens

```
Thread: main, Exception: java.lang.RuntimeException: Unable to start receiver im.vector.app.push.fcm.EmbeddedFCMDistributor: java.lang.NullPointerException
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3855)
        at android.app.ActivityThread.access$1500(ActivityThread.java:224)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1898)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:224)
        at android.app.ActivityThread.main(ActivityThread.java:7562)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
     Caused by: java.lang.NullPointerException
        at org.unifiedpush.android.embedded_fcm_distributor.EmbeddedDistributorReceiver.onReceive(EmbeddedDistributorReceiver.kt:22)
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3846)
        at android.app.ActivityThread.access$1500(ActivityThread.java:224) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1898) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:224) 
        at android.app.ActivityThread.main(ActivityThread.java:7562) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950) 
```

| GIF | 
| --- |
up-remove-crash

Comment (p1gp1g) [reply vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:107]: Thanks, I've spotted the issue, in the embedded distributor. I'm fixing it tonight

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:107]: great to hear, thanks! :raised_hands:

Comment (p1gp1g) [reply vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt:107]: done with android-embedded_fcm_distributor:2.1.2

**Pull request merged**